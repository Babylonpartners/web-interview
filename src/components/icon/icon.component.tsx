import React, { FunctionComponent } from 'react'

type IconType = {
  alias: string
  modifier?: string
}

export const Icon: FunctionComponent<IconType> = ({ alias, modifier }) => {
  return (
    <span
      className={`icon icon--${alias}${modifier ? `icon--${modifier}` : ''}`}
      title={alias}
    ></span>
  )
}
