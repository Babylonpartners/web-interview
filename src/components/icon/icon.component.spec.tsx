import React from 'react'
import renderer from 'react-test-renderer'

import { Icon } from './icon.component'

describe('Icon', () => {
  it('should match snapshot', () => {
    const component = renderer.create(
      <Icon alias="burger" modifier="no-tomato" />
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})
