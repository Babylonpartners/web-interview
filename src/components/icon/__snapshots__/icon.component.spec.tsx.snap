// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match snapshot 1`] = `
<span
  className="icon icon--burgericon--no-tomato"
  title="burger"
/>
`;
